### ISSUE TITLE
[Docs] How to properly create a global variable (singleton)?

### ISSUE THREAD

**Issue by Supatier** [GodotNativeTools/godot-d#35]

**Comment by sheepandshepherd** [comment]
[Autoload singletons](https://godot.readthedocs.io/en/3.0/getting_started/step_by_step/singletons_autoload.html) are like globals and work the same in any script language. You just need a normal script extending Node and assign it (or a saved scene with it) in the Project Settings:

![](https://godot.readthedocs.io/en/3.0/_images/addglobal.png)

Then access the node as described in the doc page: `getNode("/root/<Node Name>")`

**Comment by Supatier** [comment]
In C# you use `{set; get;}`. Does that mean you use `@property` in D?

**Comment by sheepandshepherd** [comment]
Yes. The AutoLoad menu is what makes the whole node containing those properties "global" and accessible from any script by the name you gave it. It's just a normal node script - the functions/variables don't need special syntax to make them global:  
```d
// assigned in AutoLoad menu with the name "Globals"
class Globals : GodotScript!Node
{
    @Property int globalInteger = 0;
}

// in a separate script:
class Player : GodotScript!KinematicBody
{
    @Method _ready()
    {
        Globals globals = owner.getNode("/root/Globals").as!Globals;
        globals.globalInteger = 100;
    }
}
```

There's a second way with shorter syntax - D's real global variables (variables defined outside of any class):  
```d
module globals;

int globalInteger = 0;
Node globalNode;
Ref!Texture globalTexture;
```
But these can *only* be accessed from D code. Godot itself can only access scripts.

**Comment by Supatier** [comment]
This needs to be in the readme..... :+1: 

If the readme is to long, I suggest breaking it up to multiple pages.

**Comment by Supatier** [comment]
Is it `@Property` with a capital `P` or `@property`
